### PR TITLE
electron: fix hash issue when replacing ffmpeg

### PR DESCRIPTION
**Pull request by marechal-p**
Hash computation is bogus, this commit fixes it.

Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

### CODE REVIEW

**Comment by marcdumais-work** [comment]
Having a look now

**Comment by marcdumais-work** [comment]
@marechal-p I think you'll need to rebase to get Anton's fix for Oracle Java not being available any more, so CI can pass.

**Comment by kittaakos** [comment]
Any chance of having a test case for the hash calculation?

**Comment by marcdumais-work** [comment]
@marechal-p For your consideration: 

I think the relatively expensive operation here is the download of the alternate (so called cached) `libffmpeg` library (specially with the potential GH API rate limit issue). So it makes sense to make an effort to avoid re-downloading that library unless we really need-to.

However I think that we do not need to make much effort to avoid overwriting the default version of this library that comes with Electron - that operation is really cheap. So I propose that we do not need to compute the checksums of both library to avoid the overwriting. Instead we could verify that the cached version is proprietary-codecs-free and copy it over the Electron version, every time. 

WDYT?

**Comment by marechal-p** [comment]
@marcdumais-work I considered it, but went with the checksum anyway. At least I know now how to not write a bogus algorithm for it, hopefully.

@kittaakos test added, apparently I made a change right before pushing that broke what I had fixed, everything should be good now.

File: dev-packages/electron/tests/electron-ffmpeg-lib.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const { hashFile } = require('../electron-ffmpeg-lib');
+
+function resource(...parts) {
+    return path.resolve(__dirname, 'test-resources', ...parts);
+}
+
+describe('ffmpeg utility functions', () => {
+
+    it('hashFile', async () => {
```
Comment (marcdumais-work) [inline dev-packages/electron/tests/electron-ffmpeg-lib.spec.js]: Can we have something more meaningful than "hashfile"? e.g.: "hashfile  can successfully determine that two file's content is identical or not"?

**Comment by marcdumais-work** [comment]
Retested - works well. 

The only little flaw I see but that should not matter in normal cases is if for some reason the cached library would contain the proprietary codecs, e.g. if someone manually copied the wrong version of the library under `<theia repo>/dev-packages/electron/download`, the script would not attempt to download the correct version of the library until the wrong was was removed, and the build would fail.   

I think this is acceptable.

**Comment by marechal-p** [comment]
I will merge for now and eventually come back to your comment if we have a problem :)

Thanks for the review

File: dev-packages/electron/tests/electron-ffmpeg-lib.spec.js (hunk 1 above)
Comment (marechal-p) [reply dev-packages/electron/tests/electron-ffmpeg-lib.spec.js]: Ah, since this comment wasn't part of a review, I did not see it before merging, my bad.

**Pull request merged**